source/extensions/filters/listener/http_inspector/http_inspector.cc

```diff
@@ -90,6 +104,11 @@ ParseState Filter::parseHttpHeader(absl::string_view data) {
         return ParseState::Error;
       }
 
+      if (!parsing_begin_) {
```

Comment:
Just check whether `new_data` is empty?

Reply:
The request line my be split to `GET /anything/xxxxxxxxx HTTP/1.1` and `\r\n`. When we handle the second read event, the `new_data` is empty but the request line is legitimate.